skrub/_fuzzy_join.py

```diff
@@ -55,7 +55,44 @@ def _numeric_encoding(
     scaler.fit(np.vstack((aux_array, main_array)))
     aux_array = scaler.transform(aux_array)
     main_array = scaler.transform(main_array)
-    return main_array, aux_array
+    return csr_matrix(main_array), csr_matrix(aux_array)
```

Comment:
Currently, we are forced to use sparse data because we are mixing numerical and string (which are sparse) embeddings and we need only one format..

This might change with the #558 benchmarks if we conclude that using dense arrays is more efficient.